### PR TITLE
API coverage for product salability web api calls

### PR DESCRIPTION
**Pull request by phoenix128**
### Description
Added `is-product-salable` and `is-product-salable-for-requested-qty` webapi endpoints.

Fixed constructor errors in `ProductSalabilityError` and `ProductSalableResult` causing failures when called with inventory errors.

Test web api coverage is added

### Fixed Issues (if relevant)
1. magento-engcom/msi#1825: Add Web API endpoint for StockResolverInterface and IsProductSalableForRequestedQtyInterface

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/InventorySalesApi/etc/webapi.xml [hunk 1]
```diff
@@ -19,4 +19,16 @@
...
+        <resources>
+            <resource ref="InventorySalesApi::stock"/>
+        </resources>
+    </route>
+    <route url="/V1/inventory/is-product-salable/:sku/:stockId" method="GET">
+        <service class="Magento\InventorySalesApi\Api\IsProductSalableInterface" method="execute"/>
+        <resources>
+            <resource ref="InventorySalesApi::stock"/>
+        </resources>
+    </route>
```
Comment (maghamed) [inline app/code/Magento/InventorySalesApi/etc/webapi.xml:33]: just noticed that you added web api service for already existing endpoint 
```
    <route url="/V1/inventory/is-product-salable/:sku/:stockId" method="GET">
        <service class="Magento\InventorySalesApi\Api\IsProductSalableInterface" method="execute"/>
        <resources>
            <resource ref="InventorySalesApi::stock"/>
        </resources>
    </route>
 ...
    <route url="/V1/inventory/is-product-salable/:sku/:stockId" method="GET">
        <service class="Magento\InventorySalesApi\Api\IsProductSalableInterface" method="execute"/>
        <resources>
            <resource ref="InventorySalesApi::stock"/>
        </resources>
    </route>
```

Comment (maghamed) [reply app/code/Magento/InventorySalesApi/etc/webapi.xml:33]: I think you missed it with service `\Magento\InventorySalesApi\Api\StockResolverInterface::execute`

File: app/code/Magento/InventorySalesApi/Test/Api/IsProductSalableTest.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySalesApi\Test\Api;
+
+use Magento\TestFramework\TestCase\WebapiAbstract;
+use Magento\Framework\Webapi\Rest\Request;
+
+class IsProductSalableTest extends WebapiAbstract
```
Comment (maghamed) [inline app/code/Magento/InventorySalesApi/Test/Api/IsProductSalableTest.php:13]: hmm looks like another gap, as I expected we already have web API test coverage for `IsProductSalable`

Comment (maghamed) [reply app/code/Magento/InventorySalesApi/Test/Api/IsProductSalableTest.php:13]: just to say it once again - it's very cool, that you @phoenix128  decided to spent some part of your weekend on finalizing MSI for 2.3.0 release, as you found and filled some gaps we had.

File: app/code/Magento/InventorySalesApi/etc/webapi.xml (hunk 1 above)
Comment (phoenix128) [reply app/code/Magento/InventorySalesApi/etc/webapi.xml:33]: Yes, definitely. Fixed in my last commit.

Comment (phoenix128) [reply app/code/Magento/InventorySalesApi/etc/webapi.xml]: Yes, definitely. Fixed in my last commit.

Comment (phoenix128) [reply app/code/Magento/InventorySalesApi/etc/webapi.xml:33]: Yes, definitely. Fixed in my last commit.

File: app/code/Magento/InventorySalesApi/Test/Api/IsProductSalableTest.php (hunk 2 above)
Comment (phoenix128) [reply app/code/Magento/InventorySalesApi/Test/Api/IsProductSalableTest.php:13]: We did not have before. That is the class I added yesterday togheter with `\Magento\InventorySalesApi\Test\Api\IsProductSalableForRequestedQtyTest`

**Pull request merged**